test/e2e/app-dir/app-basepath/index.test.ts

```diff
@@ -14,6 +14,11 @@ describe('app dir - basepath', () => {
     const browser = await next.browser('/base/pages-path')
     await browser.elementByCss('#to-another').click()
     await browser.waitForElementByCss('#page-2')
+
+    const logs = await browser.log()
+
+    // Expect no errors when doing a hard navigation between the routers.
+    expect(logs.find((log) => log.source === 'error')).toBeUndefined()
```

Comment:
This test seems to pass fine on canary without the changes here.

Reply:
Yes, it passes fine but there's an error in logs, where the router tried to first load the chunk, then failed and did a full refresh navigation. With this check the log is tested and it will make the test fail.

Reply:
> With this check the log is tested and it will make the test fail

I'm saying when I run this test against current canary without any other fixes/changes it passes so the test isn't capturing any errors

Reply:
I added these lines here to the latest canary (without any other change other than adding `.only`), ran the following command:

```bash
pnpm test-dev test/e2e/app-dir/app-basepath/index.test.ts
```

And this is what I get:
```
Stopped next server
destroyed next instance: 2.118s
 FAIL  test/e2e/app-dir/app-basepath/index.test.ts (24.715 s)
  app dir - basepath
    ✕ should successfully hard navigate from pages -> app (7306 ms)
    ○ skipped should support `basePath`
    ○ skipped should support Link with basePath prefixed
    ○ skipped should prefix metadata og image with basePath
    ○ skipped should prefix redirect() with basePath
    ○ skipped should render usePathname without the basePath
    ○ skipped should handle redirect in dynamic in suspense boundary routes with basePath
    ○ skipped should only make a single RSC call to the current page (/base/refresh)
    ○ skipped should only make a single RSC call to the current page (/base/refresh?foo=bar)
    ○ skipped should properly stream an internal server action redirect() with a relative URL
    ○ skipped should properly stream an internal server action redirect() with a absolute URL
    ○ skipped should redirect externally when encountering absolute URLs on the same host outside the basePath

  ● app dir - basepath › should successfully hard navigate from pages -> app

    expect(received).toBeUndefined()

    Received: {"message": "Failed to load resource: the server responded with a status of 404 (Not Found)", "source": "error"}

      19 |
      20 |     // Expect no errors when doing a hard navigation between the routers.
    > 21 |     expect(logs.find((log) => log.source === 'error')).toBeUndefined()
         |                                                        ^
      22 |   })
      23 |
      24 |   it('should support `basePath`', async () => {

      at Object.toBeUndefined (e2e/app-dir/app-basepath/index.test.ts:21:56)
```

I'm using Node.js 20.11.0 in case that's relevant (with `pnpm dev` running in the background).